letsencrypt/auth_handler.py

```diff
@@ -153,13 +153,18 @@ def _send_responses(self, achalls, resps, chall_update):
         """
         active_achalls = []
         for achall, resp in itertools.izip(achalls, resps):
+            # XXX: make sure that all achalls, including those
+            # corresponding to None or False returned from
+            # Authenticator are removed from the queue and thus avoid
+            # infinite loop
+            active_achalls.append(achall)
```

Comment:
@jdkasten, I'm really not sure about pulling this line out of the condition branch, but I otherwise run into infinite loop problems. Is that intended from the API point of view?